cocotb/regression.py

```diff
@@ -261,84 +249,49 @@ def handle_result(self, test: RunningTask) -> None:
-                                ratio_time=repr(ratio_time))
-
-        # score test
-        result_pass, sim_failed = self._score_test(self._test, test._outcome)
 
         # stop capturing log output
         cocotb.log.removeHandler(test.handler)
 
-        # Save results
-        self._store_test_result(self._test.__module__, self._test.__qualname__, result_pass,
-                                sim_time_ns, real_time, ratio_time)
-        if not result_pass:
-            self.xunit.add_failure()
-            self.failures += 1
-
-        # Fail if required
-        if sim_failed:
-            self.tear_down()
-            return
+        self._record_result(
+            test=self._test,
+            outcome=self._running_test._outcome,
+            wall_time_s=real_time,
+            sim_time_ns=sim_time_ns)
 
         self.execute()
 
-    def _init_test(self, test_func: Test) -> Optional[RunningTask]:
+    def _init_test(self, test: Test) -> Optional[RunningTask]:
         """
         Initializes a test.
 
         Records outcome if the initialization fails.
         Records skip if the test is skipped.
         Saves the initialized test if it successfully initializes.
         """
-        test_init_outcome = cocotb.outcomes.capture(test_func, self._dut)
+
+        if test.skip:
+            hilight_start = ANSI.COLOR_TEST if want_color_output() else ''
+            hilight_end = ANSI.COLOR_DEFAULT if want_color_output() else ''
+            # Want this to stand out a little bit
+            self.log.info("{}Skipping test {}/{}:{} {}".format(
+                hilight_start,
+                self.count,
+                self.ntests,
+                hilight_end,
+                test.__qualname__))
+            self._record_result(test, None, 0, 0)
+            return
+
+        test_init_outcome = cocotb.outcomes.capture(test, self._dut)
 
         if isinstance(test_init_outcome, cocotb.outcomes.Error):
-            self.log.error("Failed to initialize test %s" % test_func.name, exc_info=True)
-            self.xunit.add_testcase(name=test_func.name,
-                                    classname=test_func.__module__,
-                                    time="0.0",
-                                    sim_time_ns="0.0",
-                                    ratio_time="0.0")
-            result_pass, sim_failed = self._score_test(test_func, test_init_outcome)
-            # Save results
-            self._store_test_result(test_func.__module__, test_func.__qualname__, result_pass, 0.0, 0.0, 0.0)
-            if not result_pass:
-                self.xunit.add_failure()
-                self.failures += 1
-            # Fail if required
-            if sim_failed:
-                self.tear_down()
-                raise SimFailure("Test initialization caused a simulator failure. Shutting down.")
+            self.log.error("Failed to initialize test %s" % test.__qualname__, exc_info=True)
```

Comment:
I think this needs to be
```suggestion
            self.log.error("Failed to initialize test %s" % test.__qualname__, exc_info=test_init_outcome.error)
```
since `exc_info` pulls from the currently thrown exception. I suspect this was already broken before your change.